### PR TITLE
change: we now enable GC64 mode by default in our bundled LuaJIT build for x86_64 architectures [...]

### PR DESCRIPTION
**Pull request by thibaultcha**
[...] this can be disabled with ./configure --without-luajit-gc64.

I hereby granted the copyright of the changes in this pull request
to the authors of this openresty project.

### CODE REVIEW

File: util/configure [hunk 1]
```diff
@@ -697,6 +701,14 @@ _END_
             $luajit_xcflags .= " -DLUAJIT_ENABLE_LUA52COMPAT";
         }
 
+        if (!$no_luajit_gc64
+            && `uname -a` =~ /\bx86_64\b/
```
Comment (thibaultcha) [inline util/configure]: Note: iirc, the GC64 mode is required on ARM64.

Comment (agentzh) [reply util/configure]: I think we should use Perl's own architecture information instead of invoking `uname` which is not available everywhere.

BTW, GC64 is required on ARM64 so we do not need to worry about that at all. The LuaJIT build system enables that automatically anyway.

File: util/configure [hunk 2]
```diff
@@ -1302,6 +1314,7 @@ _EOC_
   --with-luajit-xcflags=FLAGS        Specify extra C compiler flags for LuaJIT 2.1
   --without-luajit-lua52             Turns off the LuaJIT extensions from Lua 5.2 that may break
                                      backward compatibility.
+  --without-luajit-gc64              Turns off LuaJIT GC64 mode
```
Comment (agentzh) [inline util/configure]: We should mention x86_64 here?

File: util/configure (hunk 1 above)
Comment (thibaultcha) [reply util/configure]: Should we then update existing invocations of `uname`? i.e. https://github.com/openresty/openresty/blob/feb1bc4c2a060abf48c24a5cdbe685f156e80829/util/configure#L325

Comment (agentzh) [reply util/configure]: We could though it's less an issue since it first makes sure it is on darwin.

Comment (thibaultcha) [reply util/configure]: Updated

File: util/configure (hunk 2 above)
Comment (thibaultcha) [reply util/configure]: Good point, updated

File: util/configure (hunk 1 above)
Comment (thibaultcha) [reply util/configure]: > The LuaJIT build system enables that automatically anyway.

That's helpful, thanks, I was about to get my hands on an ARM64 box to test this (still planning on doing so today or tomorrow).

Comment (agentzh) [reply util/configure]: @thibaultcha Great. Thanks! It would be awesome if you could run the tests on 32-bit arm systems at the same time, to cover corner cases as in openresty/luajit2#37.
